Tests/Integration/JsonApiMediaTypeFormatterTests.cs

```diff
@@ -506,6 +506,41 @@ public async Task PagedResult(object baseUrl, int firstPageNumber)
             }
         }
 
+        [InlineData("api/companies/paged-result-empty", 0)]
+        [InlineData("api/companies/paged-result-empty-first-page", 1)]
+        [Theory(DisplayName = "Paged result calculates correct first and last links when there are no any items in the response")]
+        public async Task PagedResultEmpty(string baseUrl, int firstPageNumber)
+        {
+            using (var server = new NewSetupJsonApiServer(new JsonApiConfiguration()))
+            {
+                var client = server.GetClient();
+                // endpoint will return empty result
+                List<string> endpoints = new List<string>()
+                {
+                    baseUrl,
+                    // lets ask for different page to be sure that it still will have correct first and last
+                    $"{baseUrl}?page[number]=3",
+                    // then ask for custom pageSize
+                    $"{baseUrl}?page[number]=3&page[size]=1",
+                    // then for custom pageSize but without pageNumber
+                    $"{baseUrl}?page[size]=1",
+                };
+
+                foreach (var endpoint in endpoints)
+                {
+                    var result = await client.GetFullJsonResponseAsync($"{baseUrl}");
+                    var resultCount = ((JArray)result.Content["data"])?.Count;
+                    var last = result.Content["links"]["last"].Value<string>();
+                    var first = result.Content["links"]["first"].Value<string>();
+                    Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+                    Assert.Equal(0, resultCount);
+                    Assert.EndsWith($"{baseUrl}?page[number]={firstPageNumber}", first);
+                    Assert.EndsWith($"{baseUrl}?page[number]={firstPageNumber}", last);
```

Comment:
Maybe it makes sense to assert that first and last are the same? In at these scenarios:

1. there are fewer results than the page size
2. there are no results (special case)

they should be the same regardless of whether you start with 0 or 1.

Reply:
Added one more unit tests and two endpoints that returns 10 items with PageSize default as 20, so it would be less than the whole page. So in that case last\first also would be the same and next would be missing